activities/Memorize.activity/js/activity.js

```diff
@@ -21,6 +21,107 @@ define(function (require) {
+
+        function generateXOLogoWithColor(color) {
+            var coloredLogo = xoLogo;
+            coloredLogo = coloredLogo.replace("#010101", color.stroke);
+            coloredLogo = coloredLogo.replace("#FFFFFF", color.fill);
+
+            return "data:image/svg+xml;base64," + btoa(coloredLogo);
+        }
+
+        function displayUsers(users) {
+            if (onUsersListChangedCallback != null) {
+                onUsersListChangedCallback(users);
+            }
+            var presenceUsersDiv = document.getElementById("presence-users");
+            var html = "<hr><ul style='list-style: none; padding:0;'>";
+            for (var key in users) {
+                html += "<li><img style='height:30px;vertical-align:middle;' src='" + generateXOLogoWithColor(users[key].colorvalue) + "'>" + users[key].name + "</li>";
+            }
+            html += "</ul>";
+            presenceUsersDiv.innerHTML = html
+        }
+
+        function findUsersInsideAllActivities(activity, users) {
+            var realUsers = [];
+
+            for (var i = 0; i < activity.users.length; i++) {
+                for (var j = 0; j < users.length; j++) {
+                    if (users[j].networkId === activity.users[i]) {
+                        realUsers.push(users[j]);
+                    }
+                }
+            }
+
+            return realUsers;
+        }
+
+        var tmpDiv = null;
+        var tmpTimeout = null;
+
+        function showQuickModal(text) {
+            if (tmpDiv !== null) {
+                tmpDiv.parentElement.removeChild(tmpDiv);
+                if (tmpTimeout !== null) {
+                    clearTimeout(tmpTimeout)
+                }
+            }
+            var div = document.createElement('div');
+            tmpDiv = div;
+            div.style.background = "#000";
+            div.style.color = "#fff";
+            div.style.bottom = "5px";
+            div.style.padding = "15px";
+            div.style.right = "5px";
+            div.style.position = "absolute";
+            div.innerHTML = text;
+            document.body.appendChild(div);
+            tmpTimeout = setTimeout(function () {
+                tmpTimeout = null;
+                div.parentElement.removeChild(div);
+                tmpDiv = null;
+            }, 3000);
+        }
+
+        presencepalette.PresencePalette.prototype.onSharedActivityUserChanged = function (msg) {
```

Comment:
I'm not a big fan of the the idea to update the PresencePalette prototype.
Change on users list should be detected with the event onSharedActivityUserChanged on the presence object, see for example https://github.com/llaske/sugarizer/blob/dev/activities/MazeWeb.activity/js/activity.js#L35

Reply:
I have identified the mistake ! Indeed, one mus t not make changes to the prototype during runtime. I have made the necessary changes by assign callback to the mentioned event.